File: Easy/Java/Plus_One(66).java
class Solution {
    public int[] plusOne(int[] digits) {
        
        int carry = 0;
        int[] result = new int[digits.length+1];
        int sum_total = 0;
        
        for(int i = digits.length-1; i >= 0; i--)
        {
            sum_total = digits[i] + ((i == digits.length-1) ? 1 : 0) + carry;
            digits[i] = sum_total % 10;
            result[i+1] = sum_total % 10;
            carry = sum_total / 10;
        }
        
        result[0] = (carry > 0) ? 1 : 0;
                
        return (carry > 0) ? result : digits;
    }
}

File: Easy/C++/Plus_One(66).cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.size();
        
        int rem = 1;
        for(int i = n - 1; i >= 0; i--) {
            
            int temp = (digits[i] + rem);
            digits[i] = temp % 10;
            rem = temp / 10;
        }
        if(rem) {
            digits.insert(digits.begin(), rem);
        }
        return digits;
            
    }
};